app/frontend/src/features/profile/actions/PendingRequest.tsx

```diff
@@ -0,0 +1,61 @@
+/* has to check whether request is sent or received
+if sent, return a simple disabled "pending request" button
+if recieved, return the response button*/
+import { SetMutationError } from "features/connections/friends";
+import useFriendRequests from "features/connections/friends/useFriendRequests";
+import React from "react";
+
+import PendingFriendRequestSent from "./PendingFriendRequestSent";
+import RespondToFriendRequestProfileButton from "./RespondToFriendRequestProfileButton";
+
+interface PendingRequestProps {
+  userId: number;
+  setMutationError: SetMutationError;
+}
+
+function CheckSentRequests(userId: number) {
+  const friendRequestsSent = useFriendRequests("sent").data;
+  const friendRequestToUser = friendRequestsSent?.find(
```

Comment:
You can determine if you've sent a request to the user through the `user.friends` field (which you get through the user object when you're on the profile page component) and so all of the logics here to figure out from list of friend request sent is unnecessary.

Reply:
Never mind, I got a bit confused by all of these logic as I thought everything is available on the `friends` field in the user object until I found that's not the case, hence I opened the issue you saw Aapeli looked at